pkg/node/node_address_linux.go

```diff
@@ -28,116 +28,141 @@ import (
 	"github.com/vishvananda/netlink"
 )
 
-// firstGlobalV4Addr returns the first IPv4 global IP of an interface, where
-// the IPs are sorted in ascending order. when intf is defined only IPs
-// belonging to that interface are considered. If preferredIP is present in the
-// IP list it is returned irrespective of the sort order. Passing intf and
-// preferredIP will only return preferredIP if it is in the IPs that belong to
-// intf. In all cases, if intf is not found all interfaces are considered.
-func firstGlobalV4Addr(intf string, preferredIP net.IP) (net.IP, error) {
+var privateIPBlocks []*net.IPNet
+
+func init() {
+	// We only care about global scope prefixes here.
+	for _, cidr := range []string{
+		"10.0.0.0/8",     // RFC1918
+		"172.16.0.0/12",  // RFC1918
+		"192.168.0.0/16", // RFC1918
+		"fc00::/7",       // IPv6 ULA
+	} {
+		_, block, _ := net.ParseCIDR(cidr)
+		privateIPBlocks = append(privateIPBlocks, block)
+	}
+}
+
+// isPublicAddr returns whether a given global IP is from
+// a public range.
+func isPublicAddr(ip net.IP) bool {
```

Comment:
and this function can be public `IsPublicAddr`

Reply:
I moved it into pkg/ip/ip.go, so therefore made it public, yep.